### DISCUSSION TITLE
Inline list in the frontmatter of MDX files is linted as a link

### DISCUSSION THREAD

**Discussion by idlist** [mdx-js/mdx#2419] [Q&A]
![image](https://github.com/mdx-js/mdx/assets/29368226/b85bc0ad-9e3b-4adb-8b48-92c7532f87b9)

The screenshot above is a list of tags, where to make it compact I wrote it as a inline list, which is valid in yaml. But the VSCode's MDX linter seems to recognize the frontmatter as the part of the main text and lint the inline list as a link (which both use `[]` as the syntax), and thus the warning.

Is there some way to disable this warning?

**Comment by wooorm** [comment]
Frontmatter doesn’t exist in markdown normally. Try it here, doesn’t work.
Frontmatter is an extension. Extensions to MDX/markdown are enabled through plugins.
See https://github.com/mdx-js/mdx-analyzer/tree/main/packages/vscode-mdx#plugins.

**Reply by wooorm** [reply]
Also, I think that error is created by vscode-markdown? I don’t think that works well with MDX. Perhaps you can turn that off for MDX.

**Comment by remcohaszing** [comment]
This issue comes from the VSCode MDX extension. By default this supports `remark-frontmatter` and `remark-gfm`, but if plugins are explicitly configured via `tsconfig.json`, these default plugins are not included. See https://github.com/mdx-js/mdx-analyzer/blob/main/packages/vscode-mdx/README.md#plugins

This warning occurs because you have configured `#/mdx/plugins` in `tsconfig.json`, but didn’t include `remark-frontmatter`.

**Reply by idlist** [reply]
I'm using MDX with Astro and checked the `tsconfig.json` Astro provided, finding that there are no `mdx` section (and I didn't set one either), so I believe `tsconfig.json` should not be the source of the warning.

I found that in VSCode's settings, ticking `markdown.validate.enabled` and setting the following

![image](https://github.com/mdx-js/mdx/assets/29368226/24a7423a-e650-4745-8ec4-197329f75867)

to `ignore` can disable the warning, so I guess the thing is the linting of frontmatter is delegated to vscode-markdown's linter, and although vscode-markdown does support frontmatter highlight, it falsely lint the frontmatter using the content linter and thus the warning?

Quick update: this setting only affects Markdown but not MDX. In MDX the squiggles still appears. So I guess I'm wrong. Disabling MDX's server does cancel the warning, though.
